students/stuartkershaw/lesson06/test_mailroom.py

```diff
@@ -0,0 +1,61 @@
+from mailroom import donors
+from mailroom import compose_email
+from mailroom import list_donors
+from mailroom import show_donor_table
+from mailroom import generate_letters
+
+from os import listdir
+from os.path import isfile, join
+
+
+def test_donors():
+    assert isinstance(donors, dict) is True
+
+
+def test_compose_email():
+    assert compose_email('Anne Hathaway') == 'Dear Anne Hathaway, '\
+        'thanks so much for your generous donation in the amount of: $6250.'
+
+
+def test_list_donors(capsys):
+    list_donors()
+    captured = capsys.readouterr()
+    assert captured.out == 'Anne Hathaway\n'\
+        'Geronimo Jackson\n'\
+        'Dingo Dogson\n'\
+        'Leo Kottke\n'\
+        'Flask McCreole\n'\
+        'Piglet Norquist\n'
+
+
+def test_show_donor_table(capsys):
+    show_donor_table()
+    captured = capsys.readouterr()
+    assert captured.out == 'Donor Name          Total Given    Num Gifts'\
+        '      Average Gift   \n'\
+        '_________________________________________________________________\n'\
+        'Anne Hathaway        2              10250          5125.0         \n'\
+        'Geronimo Jackson     3              1800           600.0          \n'\
+        'Dingo Dogson         3              550            183.33         \n'\
+        'Leo Kottke           2              200            100.0          \n'\
+        'Flask McCreole       1              1000           1000.0         \n'\
+        'Piglet Norquist      3              65             21.67          \n'
```

Comment:
You don't need to hard code out the entire output here. I would just test the header and one of the rows, with something like:
```
assert 'Piglet Norquist      3              65             21.67          \n' in captured.out
```
It can be tricky to know where to draw the line in some cases, but generally aim for testing as small a part of the generated/dynamic output as possible to still have a meaningful test.